igor-web/src/main/java/com/netflix/spinnaker/igor/scm/DeliveryConfigManifestsController.java

```diff
@@ -0,0 +1,181 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Exposes APIs to retrieve Managed Delivery declarative manifests from source control repos. */
+@RestController
+@EnableConfigurationProperties(DeliveryConfigProperties.class)
+@RequestMapping("/delivery-config")
+public class DeliveryConfigManifestsController {
+  private static final Logger log = LoggerFactory.getLogger(AbstractCommitController.class);
+
+  private final DeliveryConfigProperties configProperties;
+  private final ObjectMapper jsonMapper;
+  private final ObjectMapper yamlMapper;
+  private final Optional<StashMaster> stashMaster;
+  private final Optional<GitHubMaster> gitHubMaster;
+  private final Optional<GitLabMaster> gitLabMaster;
+  private final Optional<BitBucketMaster> bitBucketMaster;
+
+  public DeliveryConfigManifestsController(
+      Optional<DeliveryConfigProperties> configProperties,
+      Optional<StashMaster> stashMaster,
+      Optional<GitHubMaster> gitHubMaster,
+      Optional<GitLabMaster> gitLabMaster,
+      Optional<BitBucketMaster> bitBucketMaster,
+      ObjectMapper jsonMapper) {
+    this.configProperties =
+        configProperties.isPresent() ? configProperties.get() : new DeliveryConfigProperties();
+    this.stashMaster = stashMaster;
+    this.gitHubMaster = gitHubMaster;
+    this.gitLabMaster = gitLabMaster;
+    this.bitBucketMaster = bitBucketMaster;
+    this.jsonMapper = jsonMapper;
+    this.yamlMapper =
+        new ObjectMapper(new YAMLFactory())
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+  }
+
+  /**
+   * Given details about a supported git source control repository, and optional filters for a
+   * sub-directory within the repo, the file extensions to look for, and the specific git reference
+   * to use, returns a list of (potential) Managed Delivery config manifests found at that location.
+   *
+   * <p>Note that this method does not recurse the specified sub-directory when listing files.
+   */
+  @GetMapping("/manifests")
+  public List<String> listManifests(
+      @RequestParam String scmType,
+      @RequestParam final String project,
+      @RequestParam final String repository,
+      @RequestParam(required = false, defaultValue = ".") final String directory,
+      @RequestParam(required = false, defaultValue = "yml") String extension,
+      @RequestParam(required = false, defaultValue = ScmMaster.DEFAULT_GIT_REF) final String ref) {
+
+    // TODO: this started as a method to retrieve a list of keel resource manifests, but morphed
+    //  into supposedly returning only "delivery config" manifests, even though there's not really
+    //  a good way to find out if a file is a delivery config or not, even if we were to parse each
+    //  file (it doesn't have a Kubernetes-like apiVersion  and kind like other resources.
```

Comment:
er... should it though? maybe that's a separate but valuable conversation?

Reply:
like, maybe not a `kind`, but an `apiVersion` seems totally reasonable if we want to do breaking schema updates over time.

Reply:
I think so. I never quite understood why the delivery config wasn’t modeled like resources. IMHO, anything that is part of an external contract with users should have an API version and type.

Reply:
(1) we're no longer allowing individual resources, so this todo is confusing. There will be only one file type.
(2) no version on delivery configs is a super valid point and probably worth discussing - let's create an issue for it.

Reply:
@emjburns I know we’re not, but I thought we agreed to restrict that on the orca side, not igor’s. If we want to restrict it here and enforce the `spinnaker.yml` convention I’m cool with that, but then I’ll remove the list API and rename the other. BTW, I kinda think we’d be hijacking `spinnaker.yml` for a specific feature, so I would suggest `delivery.yml` instead, given the contents of the file. Thoughts on all of the above?

Reply:
If we want to restrict this on the orca side the todo is still confusing. Maybe we can have an offline discussion about the design of this controller to understand how things should be named?

I'm unsure what you mean with "I kinda think we’d be hijacking spinnaker.yml for a specific feature".

Reply:
I’ll remove the to-do, but what do you think about restricting this to retrieve one the delivery config file in Igor? I’m inclined to go that route, actually. It was unfortunate timing for me that we removed support for resources exactly while I was working on this, but I rather take a bit extra time to get it right.

About `spinnaker.yml` I just thought there might be other things about spinnaker config users might end up putting in their repos eventually and by taking that file name for MD we’d prevent any future usage, is all.

Reply:
@emjburns I've decided to leave orca in charge of specifying the manifest name and kept the method to list manifests, just in case the UI needs it and/or we eventually support multiple delivery config files in one dir.

Reply:
sounds good!